### ISSUE TITLE
igraph "assortativity" for continuous attributes return NA if all attributes are integer, but it works if you add a constant .001

### ISSUE THREAD

**Issue by JacobElder** [igraph/rigraph#486]
I tried to use igraph's assortativity function. It returns positive values if more similar nodes have similar attributes and negative values otherwise.

Indeed, if I randomly generate attributes for a graph, this is the case:

```
    gg<-random.graph.game(50,0.3,"gnp",directed=T)
    V(gg)$group<-sample(1:7,50,replace=T)
    assortativity(gg, as.numeric(V(gg)$group))
    [1] 0.002059966
    V(gg)$group
     [1] 1 4 7 2 6 5 4 3 3 5 5 1 4 1 3 3 2 4 1 3 7 7 1 1 2 3 6 2 7 4 2 4 1 6 1 2 2 5 2 1 7 1 1 3 6 1 3 6 7 2
```

This all looks right and makes sense. However, for my dataset, I have a network for a particular situation where all the attributes are identical. Theoretically, this should be high/perfect homophily/assortativity (i.e., 1). But to the contrary, it returns NaN.

```
    V(gg)$group<-7
    assortativity(gg, as.numeric(V(gg)$group))
    [1] NaN
```

However, I have found a solution to this, although I cannot explain why this is the case. It seems to be something about the whole number/integer. If I add .001 as a constant to each number in the graph's attribute, to get 7.001 instead of 7, it returns the expected assortativity coefficient of "1". See below:

```
    gg<-random.graph.game(50,0.3,"gnp",directed=T)
    V(gg)$group<-7
    assortativity(gg, as.numeric(V(gg)$group)+rnorm(1,0,.001))
    [1] 1
```

In other words, assortativity cannot compute perfect homophily for integers/whole numbers but it can for float/numeric numbers with just a .001 added on.
